### ISSUE TITLE
Early Hints support

### ISSUE THREAD

**Issue by SuperOleg39** [envoyproxy/envoy#25881]
*Title*: Is Early Hints supported?

*Description*:

With Early Hints you can send multiple 103 responses to the client:

```
HTTP/1.1 103 Early Hints
Link: </main.css>; rel=preload; as=style

HTTP/1.1 103 Early Hints
Link: </style.css>; rel=preload; as=style
Link: </script.js>; rel=preload; as=script

HTTP/1.1 200 OK
Date: Fri, 26 May 2017 10:02:11 GMT
Content-Length: 1234
Content-Type: text/html; charset=utf-8
Link: </main.css>; rel=preload; as=style
Link: </newstyle.css>; rel=preload; as=style
Link: </script.js>; rel=preload; as=script
```

Looks like that 103 headers is ignored, and response proxied only after 200 status.
Maybe this filter responsible for this behaviour - https://github.com/envoyproxy/envoy/blob/main/contrib/golang/filters/http/source/go/pkg/api/type.go#L38 (don't sure this code are relates to problem)

*Relevant Links*

Early Hints specification - https://httpwg.org/specs/rfc8297.html#early-hints

**Comment by doujiang24** [comment]
> Maybe this filter responsible for this behaviour - https://github.com/envoyproxy/envoy/blob/main/contrib/golang/filters/http/source/go/pkg/api/type.go#L38 (don't sure this code are relates to problem)

I don't think this code is related to the problem, as the code owner of this extension. but I have no idea about the root issue.

**Comment by SuperOleg39** [comment]
By default all 1xx headers must be proxied to the client?

**Comment by zuercher** [comment]
The code you linked doesn't refer to an HTTP status code. It's just an enumeration whose value happens to be 103.

There is at least some support for status code 103: https://github.com/envoyproxy/envoy/blob/main/test/common/http/http1/codec_impl_test.cc#L2782

Perhaps @alyssawilk can comment?

**Comment by alyssawilk** [comment]
yeah Envoy will proxy one 1xx but due to [implementation details] currently only proxies 1 set of 1xx headers.  We know this isn't ideal but no one has signed on to allow multiple 103s, or multiple 1xx of any combination so that's where we are today.

**Comment by SuperOleg39** [comment]
Thanks for the answers!

Will be watching for https://github.com/envoyproxy/envoy/issues/19044

**Comment by SuperOleg39** [comment]
> Looks like that 103 headers is ignored, and response proxied only after 200 status.

Maybe it's the specifics of our k8s infrastructure, will research it
